en/lessons/basics/pattern-matching.md

```diff
@@ -1,107 +1,283 @@
 ---
-version: 1.0.2
+version: 1.0.0
 title: Pattern Matching
 ---
 
-Pattern matching is a powerful part of Elixir. It allows us to match simple values, data structures, and even functions.
-In this lesson we will begin to see how pattern matching is used.
+Haskell offers a powerful type-based system for defining functions called 
+*pattern matching*
+
+In general, pattern matching is used to deconstruct algebraic types
+
+We will look at an example of how it works, talk about it in general, then see
+some more examples
 
 {% include toc.html %}
 
-## Match Operator
+## Extensions and Imports
+
+You will need to add the following lines to the top of your Haskell source 
+file (`.hs`) to be able to run these examples
+
+```haskell
+{-# LANGUAGE OverloadedStrings #-}
+
+import Data.Text ( Text )
```

Comment:
Same remark as for the previous document above: Why make Text the default instead of String?